vllmd-hypervisor/generate-debian-image.sh

```diff
@@ -0,0 +1,320 @@
+    local initrd_url="http://ftp.debian.org/debian/dists/${DEBIAN_VERSION}/main/installer-amd64/current/images/netboot/debian-installer/amd64/initrd.gz"
+    
+    if [[ "${DRY_RUN}" -eq 0 ]]; then
+        echo "Downloading Debian ${DEBIAN_VERSION} netboot installer..."
+        wget -O "${BUILD_DIR}/kernel" "${kernel_url}" || { echo "Failed to download kernel"; exit 1; }
+        wget -O "${BUILD_DIR}/initrd.gz" "${initrd_url}" || { echo "Failed to download initrd"; exit 1; }
+        echo "Download completed."
+    else
+        echo "[DRY RUN] Would download Debian ${DEBIAN_VERSION} netboot installer from:"
+        echo "  - ${kernel_url}"
+        echo "  - ${initrd_url}"
+    fi
+}
+
+# Create preseed configuration disk
+create_preseed_disk() {
+    local preseed_disk="${BUILD_DIR}/preseed.img"
+    local temp_dir=$(mktemp -d)
+    
+    if [[ "${DRY_RUN}" -eq 0 ]]; then
+        echo "Creating preseed configuration disk..."
+        
+        # Copy preseed file to temporary directory
+        cp "${PRESEED_PATH}" "${temp_dir}/preseed.cfg"
+        
+        # Create FAT filesystem image for preseed disk
+        /usr/sbin/mkdosfs -n PRESEED -C "${preseed_disk}" 8192
+        
+        # Copy preseed file to the image
+        mcopy -oi "${preseed_disk}" -s "${temp_dir}/preseed.cfg" ::
+        
+        echo "Preseed configuration disk created at ${preseed_disk}"
+        rm -rf "${temp_dir}"
+    else
+        echo "[DRY RUN] Would create preseed configuration disk at ${preseed_disk}"
+        echo "[DRY RUN] Would copy preseed file from ${PRESEED_PATH}"
+    fi
+}
+
+# Create the VM disk image
+create_disk_image() {
+    if [[ "${DRY_RUN}" -eq 0 ]]; then
+        echo "Creating disk image for VM..."
+        # Create an empty file of the specified size using truncate (no QEMU dependency)
+        truncate -s "${DISK_SIZE}" "${OUTPUT_PATH}"
+        echo "Disk image created at ${OUTPUT_PATH}"
+    else
+        echo "[DRY RUN] Would create ${DISK_SIZE} disk image at ${OUTPUT_PATH}"
+    fi
+}
+
+# Run the installation using cloud-hypervisor
+run_installation() {
+    local kernel_path="${BUILD_DIR}/kernel"
+    local initrd_path="${BUILD_DIR}/initrd.gz"
+    local preseed_disk="${BUILD_DIR}/preseed.img"
+    local kernel_args="console=ttyS0 auto=true priority=critical interface=auto url=file:///preseed.cfg"
+    
+    if [[ "${DRY_RUN}" -eq 0 ]]; then
+        echo "Starting Debian installation using cloud-hypervisor..."
+        echo "This will take some time. Installation logs will be displayed."
+        
+        # Run cloud-hypervisor with the specified configuration
+        cloud-hypervisor \
+            --kernel "${kernel_path}" \
+            --initramfs "${initrd_path}" \
+            --cmdline "${kernel_args}" \
+            --memory "size=${MEMORY_SIZE}" \
+            --disk "path=${OUTPUT_PATH},readonly=off" \
+            --disk "path=${preseed_disk},readonly=on" \
+            --console tty \
+            --serial tty
```

Comment:
it will be hard to install debian without a network! Use this code to setup the macvlan tap interface:
```
ip link add link \"\${VLLMD_HYPERVISOR_HOST_NET}\" name macvtap\${VLLMD_KVM_ID} type macvtap'
ip link set macvtap\${VLLMD_HYPERVISOR_ID} up
ip link delete macvtap\${VLLMD_HYPERVISOR_ID}

```

When launching cloud-hypervisor:
```
--net fd=3,num_queues=8 3<>/dev/tap\$(cat /sys/class/net/macvtap\${VLLMD_HYPERVISOR_ID}/ifindex)
```

Expalin why I suggest this code and explain what the code does, and explain why.